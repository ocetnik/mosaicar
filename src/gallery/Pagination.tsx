import * as React from "react";

interface IPaginationProps {
    onPrevPageClick: () => void;
    onNextPageClick: () => void;
    pageNumber: number;
}

class Pagination extends React.Component<IPaginationProps, {}> {
    public render() {
        return (
            <div>
                <button
                    onClick={this.props.onPrevPageClick}
                    disabled={this.props.pageNumber === 1}>
                    Previous page
                </button>
                <span> Page: {this.props.pageNumber} </span>
                <button
                    onClick={this.props.onNextPageClick}>
                    Next page
                </button>
            </div>
        );
    }
}

export default Pagination;
