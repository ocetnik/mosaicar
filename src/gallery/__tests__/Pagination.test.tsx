import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Pagination from '../Pagination';

describe('Pagination', () => {
    const mockOnPrevPageClick = jest.fn();
    const mockOnNextPageClick = jest.fn();

    test('Back button is disabled when pageNumber=1', () => {
        const component = renderer.create(
            <Pagination
                onPrevPageClick={mockOnPrevPageClick}
                onNextPageClick={mockOnNextPageClick}
                pageNumber={1}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Back button is enabled when pageNumber=2', () => {
        const component = renderer.create(
            <Pagination
                onPrevPageClick={mockOnPrevPageClick}
                onNextPageClick={mockOnNextPageClick}
                pageNumber={2}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
