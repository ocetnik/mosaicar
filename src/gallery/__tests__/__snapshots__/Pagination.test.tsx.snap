// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Back button is disabled when pageNumber=1 1`] = `
<div>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Previous page
  </button>
  <span>
     Page: 
    1
     
  </span>
  <button
    onClick={[Function]}
  >
    Next page
  </button>
</div>
`;

exports[`Pagination Back button is enabled when pageNumber=2 1`] = `
<div>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Previous page
  </button>
  <span>
     Page: 
    2
     
  </span>
  <button
    onClick={[Function]}
  >
    Next page
  </button>
</div>
`;
